compression.go

```diff
@@ -32,7 +33,8 @@ func (c Compression) newReader(r io.Reader) (io.ReadCloser, error) {
 func (c Compression) newWriter() compressedWriter {
 	switch c {
 	case CompressionGzip:
-		return new(gzip.Writer)
+		wr, _ := gzip.NewWriterLevel(ioutil.Discard, gzip.BestSpeed)
```

Comment:
Thanks for this, good point and much appreciated, but what's the rationale for `BestSpeed`? I assume someone who's using compression is more concerned with size than speed, wouldn't you agree? I would suggest `DefaultCompression` instead.

Reply:
DefaultCompression would be fine as a balance between speed and space, and in my experience there isn't that much difference when compared to BestSpeed. I guess I was thinking that given the compressed files were just an intermediate step (and not the final result), along with the fact that the compressed size will vary greatly with input anyway, the caller would most likely appreciate a little extra performance more than saving a little space that they were probably not even aware of.

Regardless, I like your final PR. Thanks for the quick turnaround.